app/controllers/user_credentials_controller.rb

```diff
@@ -14,6 +14,24 @@ def index
     end
   end
 
+  def import
+    file = params[:file]
+    content = JSON.parse(file.read)
+    new_credentials = content.map do |hash|
+      current_user.user_credentials.build(hash.slice("credential_name", "credential_value", "mode"))
+    end
+
+    if new_credentials.map(&:save).all? { |save_status| save_status }
```

Comment:
Is the block needed? `save` should return `true` or `false`, so `all?` should work without it.

You could also move the `if` conditional inside the `respond_to` block which would remove two lines of code :smile: